mixins/declarative-event-listeners.html

```diff
@@ -0,0 +1,79 @@
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="../../polymer/lib/utils/boot.html">
+<link rel="import" href="../../polymer/lib/utils/mixin.html">
+
+<script>
+  (function () {
+    'use strict';
+    /**
+     * @const {Polymer.DeclarativeEventListeners}
+     */
+    const declarativeListeners = Polymer.DeclarativeEventListeners;
+    /**
+     * Element class mixin that provides API for adding declarative event
+     * listeners nodes.
+     *
+     * The API is designed to be compatible with override points implemented
+     * in `Polymer.TemplateStamp` such that declarative event listeners in
+     * templates will support gesture events when this mixin is applied along with
+     * `Polymer.TemplateStamp`.
+     *
+     * @mixinFunction
+     * @polymer
+     * @memberof Polymer
+     * @summary Element class mixin that provides API for adding declarative event
+     * listeners nodes
+     */
+    Polymer.DeclarativeEventListeners = Polymer.dedupingMixin(superClass => {
+
+      /**
+     * @polymer
+     * @mixinClass
+     * @implements {Polymer_DeclarativeEventListeners}
```

Comment:
These interfaces for JS Compiler are generated by https://github.com/Polymer/gen-closure-declarations. Maybe we should be running that and including an `externs.js` here. Same for TypeScript via https://github.com/Polymer/gen-typescript-declarations.

Don't worry about it for now though, I can hook that up after this PR is in.